### PR TITLE
feat(ingress): Use nginx-route

### PR DESCRIPTION
**Pull request by merkata**
As the ingress relation is getting deprecated, this PR removes that relation in favor of the new nginx-route relation.

### CODE REVIEW

File: lib/charms/nginx_ingress_integrator/v0/nginx_route.py [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+- charm: (the charm itself)
+- service_hostname
+- service_name
+- service_port
+
+Other optional arguments include:
+- additional_hostnames
+- backend_protocol
+- limit_rps
+- limit_whitelist
```
[the remark this answers is not in the document]
Comment (gregory-schiano) [reply lib/charms/nginx_ingress_integrator/v0/nginx_route.py:18]: You can add it to ignore list, this needs to be tackled on nginx-ingress-integrator side

**Review by gregory-schiano — COMMENTED** [review]
Code change looks good to me, let's see if integration tests are fine and add the list to ignore list of `woke` tool

File: lib/charms/nginx_ingress_integrator/v0/nginx_route.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (arturo-seijas) [reply lib/charms/nginx_ingress_integrator/v0/nginx_route.py:18]: IT's the nginx terminology, so it'd probably be better to keep it as it is

**Review by jdkandersson — DISMISSED** [review]

**Review by arturo-seijas — DISMISSED** [review]
LGTM

**Review by jdkandersson — DISMISSED** [review]

**Review by amandahla — APPROVED** [review]
LGTM

**Review by yanksyoon — APPROVED** [review]
LGTM!

**Pull request merged**